sys/riscv/include/atomic.h

```diff
@@ -512,28 +374,28 @@ static __inline void
 atomic_thread_fence_acq(void)
 {
 
-	fence();
+	fence(); /* TODO: __atomic_thread_fence(__ATOMIC_ACQUIRE) */
```

Comment:
You should be able to use the commented version for the fences.  They are intended to match C11 semantics.